modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala

```diff
@@ -44,155 +44,126 @@ import scala.concurrent.{ExecutionContext, Future}
-            traceName("searchInstancesOfOrg") {
-              val filter = filterFrom(deprecatedOpt, filterOpt, querySettings.nexusVocBase)
-              instanceQueries.list(orgId, filter, pagination, termOpt).buildResponse(base, pagination)
-            }
+        extractAnyResourceId() { id =>
+          (pathEndOrSingleSlash & resourceId(id, of[OrgId])) { orgId =>
+            instanceQueries.list(orgId, filter, pagination, termOpt).buildResponse(base, pagination)
+          } ~
+            (pathEndOrSingleSlash & resourceId(id, of[DomainId])) { domainId =>
+            instanceQueries.list(domainId, filter, pagination, termOpt).buildResponse(base, pagination)
+          } ~
+          (pathEndOrSingleSlash & resourceId(id, of[SchemaName])) { case SchemaName(domainId, name) =>
+            instanceQueries.list(domainId, name, filter, pagination, termOpt).buildResponse(base, pagination)
+          } ~
+          (pathEndOrSingleSlash & resourceId(id, of[SchemaId])) { schemaId =>
+            instanceQueries.list(schemaId, filter, pagination, termOpt).buildResponse(base, pagination)
+          } ~
+          (resourceId(id, of[InstanceId]) & pathPrefix("outgoing") & pathEndOrSingleSlash) { instanceId =>
+            instanceQueries.outgoing(instanceId, filter, pagination, termOpt).buildResponse(base, pagination)
+          } ~
+          (resourceId(id, of[InstanceId]) & pathPrefix("incoming") & pathEndOrSingleSlash) { instanceId =>
+            instanceQueries.incoming(instanceId, filter, pagination, termOpt).buildResponse(base, pagination)
+          }
+        }
+      }
+    }
 
+  protected def resourceRoutes: Route =
+    extractAnyResourceId() { id =>
+      (pathEndOrSingleSlash & post & resourceId(id, of[SchemaId])) { schemaId =>
+        entity(as[Json]) { json =>
+          traceName("createInstance") {
+            onSuccess(instances.create(schemaId, json)) { ref =>
+              complete(StatusCodes.Created -> ref)
+            }
+          }
+        }
+      } ~
+      resourceId(id, of[InstanceId]) { instanceId =>
+        pathEndOrSingleSlash {
+          (put & entity(as[Json]) & parameter('rev.as[Long])) { (json, rev) =>
+            traceName("updateInstance") {
+              onSuccess(instances.update(instanceId, rev, json)) { ref =>
+                complete(StatusCodes.OK -> ref)
+              }
+            }
           } ~
-          pathPrefix(Segment) { domain =>
-            val domainId = DomainId(orgId, domain)
-            (pathEndOrSingleSlash & get & searchQueryParams) { (pagination, filterOpt, termOpt, deprecatedOpt) =>
-                traceName("searchInstancesOfDomain") {
-                  val filter = filterFrom(deprecatedOpt, filterOpt, querySettings.nexusVocBase)
-                  instanceQueries.list(domainId, filter, pagination, termOpt).buildResponse(base, pagination)
+          (get & traceName("getInstance")) {
```

Comment:
I think it's useful to discriminate between getInstance and getInstanceRevision like before this change. Thoughts?

Reply:
It is fine. I'll then revert that block